tests/client/test_proxies.py

```diff
@@ -87,3 +88,18 @@ def test_dispatcher_for_request(url, proxies, expected):
 def test_unsupported_proxy_scheme():
     with pytest.raises(ValueError):
         httpx.AsyncClient(proxies="ftp://127.0.0.1")
+
+
+def test_no_proxy_returns_correct_dispatcher(monkeypatch):
+    monkeypatch.setenv("HTTP_PROXY", "http://example.com")
+    monkeypatch.setenv("NO_PROXY", "google.com")
+    client = httpx.AsyncClient()
+    dispatcher = client.dispatcher_for_url(URL("http://google.com"))
+    assert dispatcher == client.dispatch
+
+
+def test_no_proxy_not_set_returns_correct_dispatcher(monkeypatch):
+    monkeypatch.setenv("HTTP_PROXY", "http://example.com")
+    client = httpx.AsyncClient()
+    dispatcher = client.dispatcher_for_url(URL("http://google.com"))
+    assert dispatcher == client.proxies["http"]
```

Comment:
I think we can use the style from the other test cases here, and parametrize this test to be able to easily add more test cases if needed (e.g. I added one below for the default "no env set" case), so…

```suggestion
@pytest.mark.parametrize(
    ["url", "env", "expected"],
    [
        ("http://google.com", {}, None),
        (
            "http://google.com",
            {"HTTP_PROXY": "http://example.com"},
            "http://example.com",
        ),
        (
            "http://google.com",
            {"HTTP_PROXY": "http://example.com", "NO_PROXY": "google.com"},
            None,
        ),
    ],
)
def test_proxies_environ(monkeypatch, url, env, expected):
    for name, value in env.items():
        monkeypatch.setenv(name, value)

    client = httpx.AsyncClient()
    dispatcher = client.dispatcher_for_url(httpx.URL(url))

    if expected is None:
        assert dispatcher == client.dispatch
    else:
        assert dispatcher.proxy_url == expected
```